integration-tests/features/src/s3interface.py

```diff
@@ -139,3 +141,23 @@ def read_object_metadata(self, bucket_name, key, attribute):
     def selector_to_key(selector):
         """Construct a key from given selector (that is written in tests w/o underscores)."""
         return selector.lower().replace(" ", "_")
+
+    def get_object_from_s3(self):
+        """Download Manifest Files from S3."""
+        base_url = os.environ.get('MANIFESTS_BUCKET_URL', None)
+        assert base_url is not None, 'Please set MANIFESTS_BUCKET_URL in environment'
+        files_to_download = ['npmlist.json', 'dependencies.json', 'pylist.json']
+        path = 'dynamic_manifests/'
+        if not os.path.exists(path):
+            os.makedirs(path)
+
+        for obj in files_to_download:
+            save_path = base_url + path + obj
+            try:
+                filedata = urlrequest.urlopen(url=save_path).read()
+                with open(path + obj, 'wb') as f:
+                    f.write(filedata)
+            except botocore.exceptions.ClientError:
+                print("The object does not exist.", obj)
+                return 'error in file fetching from s3', 400
+        return 'success', 200
```

Comment:
why do you return something like HTTP status code. Is not better to raise exception or (at least) return flag whether the operation is ok or not. Is this function copied from another source code, right?

Reply:
@tisnik If we can see caller func. https://github.com/deepak1725/fabric8-analytics-common/blob/6ff59c8ab7a0ea8383c974cc53faf75fbb9f7d27/integration-tests/features/steps/stack_analysis.py#L818

Here assert will trigger only when format remains the same in success and fail scenerios.

Also i am Okay to update it to raise Exception here. Please suggest